### PR TITLE
WT-9251 Perform a log flush by checkpoint transaction commit

### PR DESCRIPTION
**Pull request by kommiharibabu**
Checkpoint may include the updates from a recent transactions
due to some parallel eviction activity. Ensure that all the log
writes that are performed before metadata gets checkpointed
as part of the checkpoint commit transaction to ensure that all
the log reecords are written to the disk before the checkpoint
is become stable.

### CODE REVIEW

File: dist/stat_data.py [hunk 1]
```diff
@@ -579,6 +579,7 @@ def __init__(self, name, desc, flags=''):
     TxnStat('txn_checkpoint_time_min', 'transaction checkpoint min time (msecs)', 'no_clear,no_scale'),
     TxnStat('txn_checkpoint_time_recent', 'transaction checkpoint most recent time (msecs)', 'no_clear,no_scale'),
     TxnStat('txn_checkpoint_time_total', 'transaction checkpoint total time (msecs)', 'no_clear,no_scale'),
+    TxnStat('txn_checkpoint_running_timing_stress_stop', 'transaction checkpoint currently running at checkpoint stop timing stress', 'no_clear,no_scale'),
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply dist/stat_data.py]: Thanks. Fixed.

File: src/conn/conn_log.c [hunk 2]
```diff
@@ -310,6 +310,10 @@ __wt_logmgr_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfig)
     if (cval.val != 0)
         conn->log_prealloc = 1;
 
+    WT_RET(__wt_config_gets(session, cfg, "log.force_write_wait", &cval));
+    if (cval.val != 0)
+        conn->txn_log_force_write_wait = (uint32_t)cval.val;
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/conn/conn_log.c]: Removed.

File: src/txn/txn.c [hunk 3]
```diff
@@ -1571,6 +1571,14 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
              */
         }
 
+        /*
+         * Checkpoint commit transaction must perform log sync irrespective of connection level
+         * setting.
+         */
+        if (!FLD_ISSET(txn->txn_logsync, WT_LOG_DSYNC | WT_LOG_FSYNC) &&
+          WT_SESSION_IS_CHECKPOINT(session))
+            txn->txn_logsync = WT_LOG_FSYNC;
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/txn/txn.c]: Checkpoint transaction changes to the metadata to update each btree checkpoint are logged. So I thought of letting this transaction commit will fix the problem. 

I changed it as per your suggestion to let the checkpoint flush all the logs that are written after the checkpoint transaction gets committed, as the flush is related to the checkpoint and having it in txn_ckpt.c seems good.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Review by sueloverso — APPROVED** [review]
LGTM with one minor point.

File: src/include/connection.h [hunk 4]
```diff
@@ -485,6 +485,7 @@ struct __wt_connection_impl {
     uint32_t log_prealloc;                 /* Log file pre-allocation */
     uint16_t log_req_max;                  /* Max required log version */
     uint16_t log_req_min;                  /* Min required log version */
+    uint32_t log_force_write_wait;         /* Log force write wait configuration */
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/include/connection.h]: Thanks, fixed.

**Review by agorrod — APPROVED** [review]
Sorry for the delay, lgtm

**Pull request merged**